apps/examples/testcase/le_tc/kernel/tc_pthread.c

```diff
@@ -1539,6 +1539,60 @@ static void tc_pthread_pthread_setcanceltype(void)
 }
 #endif
 
+#ifndef CONFIG_PTHREAD_MUTEX_UNSAFE
+static void *test_mutex_consistent(void *param)
+{
+	int ret = -1;
+	ret = pthread_mutex_lock(&g_mutex);
+	return (void *)ret;
+}
+
+/**
+* @fn                   :tc_pthread_pthread_mutex_consistent
+* @brief                :This tc tests pthread_mutex_consistent()
+* @Scenario             :the pthread_mutex_consistent() function can be used to mark the state protected by the mutex referenced by mutex as consistent again
+*                        and this situation arises, If an owner of a robust mutex terminates while holding the mutex,
+*                        the mutex becomes inconsistent and the next thread that acquires the mutex lock shall be notified of the state.
+*                        In this case, the mutex does not become normally usable again until the state is marked consistent.
+* @API'scovered         :pthread_mutex_consistent
+* @Preconditions        :none
+* @Postconditions       :none
+* @return               :void
+*/
+static void tc_pthread_pthread_mutex_consistent(void)
+{
+	int ret;
+	void *status;
+	pthread_t thread1;
+	pthread_t thread2;
+	static pthread_mutexattr_t mutex_attr;
+
+	pthread_mutexattr_init(&mutex_attr);
+	pthread_mutexattr_setrobust(&mutex_attr, PTHREAD_MUTEX_ROBUST);
+	pthread_mutex_init(&g_mutex, &mutex_attr);
+
+	/* creates the 1st thread to lock mutex */
+	ret = pthread_create(&thread1, NULL, &test_mutex_consistent, NULL);
+	pthread_join(thread1, &status);
+	TC_ASSERT_EQ_CLEANUP("pthread_create", ret, 0, pthread_mutex_unlock(&g_mutex));
```

Comment:
If pthread is not created well, we don't need to call `pthread_join()`. 
So I think we can move this line before calling `pthread_join()`.